example/otel-collector/aggregator_factory.go

```diff
@@ -0,0 +1,27 @@
+package main
+
+import (
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/metric/number"
+	"go.opentelemetry.io/otel/sdk/metric/aggregator/histogram"
+)
+
+type histogramAggregatorFactory struct {
+	metric.AggregatorFactory
+
+	boundaries []float64
+	descriptor metric.Descriptor
+}
+
+func NewHistogramAggregatorFactory(boundaries []float64) *histogramAggregatorFactory {
+	agg := &histogramAggregatorFactory{
+		boundaries: boundaries,
+		descriptor: metric.NewDescriptor("dummy", metric.ValueObserverInstrumentKind, number.Float64Kind),
```

Comment:
doesn't quite feel right. I know this would work for all synchronous instruments just because I checked the implementation of `histogram.New`.

Reply:
I think we have to add a new API in the `histogram` package as well.

Reply:
just checked the histogram again, looks like we still need to include the `NumberKind` to get a correct aggregation anyway.

Reply:
Yea `NumberKind` seems to be important.

Adding a new API would work, but we would have to make assumptions on the `NumberKind` like I did here for the histogram factory. Alternatively I could add two functions

```
NewFloat(opts ...Option)

NewInt(opts ...Option)
```

to cover both the cases and remove the dependency.

Reply:
the goal is to forbid people to misuse the View API, for example, registering a view with float instruments with integer aggregators.

Reply:
to achieve this, we may have to change the View API design a bit if have to.

Reply:
let me update the base branch to make the APIs more robust.

Reply:
Does it make sense to have histograms as a separate view case i.e. 

```
func NewHistogramView(inst SyncImpl, labelConfig ViewLabelConfig, labelKeys []label.Key, boundaries []float64) View {
	return View {
		inst: inst,
		labelConfig: labelConfig,
		labelKeys: labelKeys,
		aggregatorFactory: //default aggregator factory that takes into account inst.descriptor.numkind
	}
}
```

or is this adding too much implementation to the view api.

Reply:
such kind of API is not extensible.